kliff/calculators/calculator.py

```diff
@@ -364,6 +372,17 @@ def get_calculator_list(self):
             calc_list.extend([calc] * N)
         return calc_list
 
+    def has_opt_params_bounds(self):
+        """
+        Return a boolean to indicate whether there are parameters whose bounds are
+        provided.
+        """
+        calc_list = self.get_calculator_list()
```

Comment:
Can directly use3 `self._calculators`. This can make it a bit faster.

`self.get_calculator_list()` is nothing but repeat `self._calculators` multiple times.